### PR TITLE
MAUI-2286 : Prepare UG document for Maui Maps control

### PR DESCRIPTION
**Pull request by Shalini-Ashokan**
Hi @vigneshrameshsync ,

I have prepared UG content for Maui maps control. Could you please review it and let me know if you have any concerns?

Regrads,
Shalini.

### CODE REVIEW

File: MAUI/Maps/data-labels.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    }
+     public String Continent { get; set; }
+     public String ContinentCode { get; set; }
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Data labels support](images/data-labels/default-data-labels.png)
```
Comment (vigneshrameshsync) [inline MAUI/Maps/data-labels.md:78]: I have asked you to don't use a world map with continent data. The output will image is not fine. Please change it.

**Review by vigneshrameshsync — CHANGES_REQUESTED** [review]

File: MAUI/Maps/bubble.md [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+
+{% endtabs %}
+
+![Bubble tooltip](images/bubble/bubble-tooltip.png)
+
+## Color
+
+You can customize the bubble color based on the value returned from the [`MapBubbleSettings.ColorMappings`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/bubbleColorValueMapper.html) property. 
+
+If [`ColorMappings`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/bubbleColorValueMapper.html) returns a color, then the color will be applied to the bubble straightaway.
```
Comment (vigneshrameshsync) [inline MAUI/Maps/bubble.md:181]: If `ColorValuePath` provides a color, then the color will be applied to the bubble straightaway.

File: MAUI/Maps/selection.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+title: Shape Selection in MAUI Maps widget | Syncfusion
+description: Learn here all about the Shape Selection feature of Syncfusion MAUI Maps (SfMaps) widget and more.
+platform: MAUI
+control: SfMaps
+documentation: ug
+---
+
+# Shape Selection in MAUI Maps (SfMaps)
+
+You can select a shape in order to highlight that area on a map. You can use the callback for performing any action during shape selection.
```
Comment (vigneshrameshsync) [inline MAUI/Maps/selection.md:12]: callback -> event

File: MAUI/Maps/selection.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+## Enable shape selection
+
+You can enable shape selection on a map using the [`MapShapeLayer.EnableSelection`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeLayer/onSelectionChanged.html) property.Selection allows you select only one shape at a time. You can select a shape by tapping it. By default, the selection is enabled when the EnableSelection property is set to true.
+
+The [`ShapeSelected`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeLayer/onSelectionChanged.html) callback is used to pass the index of the selected shape when the user is selecting a shape by tapping or clicking or by programmatically.
```
Comment (vigneshrameshsync) [inline MAUI/Maps/selection.md:18]: We didn't pass the index value in the `ShapeSelected` event.

File: MAUI/Maps/selection.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        </map:SfMaps.Layer>
+    </map:SfMaps>
+
+{% endhighlight %}
+
+{% highlight c# %}
+
+private void MapShapeLayer_ShapeSelected(object sender, ShapeSelectedEventArgs e)
+{
+     layer.SelectedShapeFill = Colors.Aqua;
```
Comment (vigneshrameshsync) [inline MAUI/Maps/selection.md:38]: Please remove it and set the color from XAML itself.

File: MAUI/Maps/selection.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+![Enable shape selection](images/selection/enable-shape-selection.png)
+
+## Appearance customization
+
+You can customize the below appearance of the selected shape.
+
+* **SelectedShapeFill** - Change the background color of the selected shape using the [`MapShapeLayer.SelectedShapeFill`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapSelectionSettings/color.html) property.
+* **SelectedShapeStrokeThickness** - Change the stroke width of the selected shape using the [`MapShapeLayer.SelectedShapeStrokeThickness`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapSelectionSettings/strokeWidth.html) property.
+* **SelectedShapeStroker** - Change the stroke color of the selected shape using the [`MapShapeLayer.SelectedShapeStroke`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapSelectionSettings/strokeColor.html) property.
```
Comment (vigneshrameshsync) [inline MAUI/Maps/selection.md:53]: SelectedShapeStroker -> SelectedShapeStroke

File: MAUI/Maps/selection.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+{% endtabs %}
+
+![Enable shape selection](images/selection/enable-shape-selection.png)
+
+## Appearance customization
+
+You can customize the below appearance of the selected shape.
+
+* **SelectedShapeFill** - Change the background color of the selected shape using the [`MapShapeLayer.SelectedShapeFill`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapSelectionSettings/color.html) property.
```
Comment (vigneshrameshsync) [inline MAUI/Maps/selection.md:51]: If SelectedShapeFill is null then the saturated color of the shape will be applied.
If SelectedShapeFill is Transparent, then the UI changes will not occur.

File: MAUI/Maps/tooltip.md [hunk 8]
```diff
@@ -0,0 +1,468 @@
...
+{% endtabs %}
+
+![Maps marker tooltip builder](images/tooltip/marker_tooltip_builder.png)
+
+N>
+* Refer the [`MapTooltipSettings`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapTooltipSettings-class.html), for customizing the tooltip.
+
+## Appearance customization
+
+## Customizing Data Templates in ToolTip
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tooltip.md:251]: Provide it as H3 heading.
Check the casing

File: MAUI/Maps/tooltip.md [hunk 9]
```diff
@@ -0,0 +1,468 @@
...
+
+Data template can be customized for items and headers of the control. The following code example shows the usage of DataTemplate.
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tooltip.md:253]: Please refer to the below UG for descriptions

https://help.syncfusion.com/maui/listview/viewappearance

File: MAUI/Maps/tooltip.md [hunk 10]
```diff
@@ -0,0 +1,468 @@
...
+
+{% tabs %}
+
+{% highlight xml %}
+
+ <map:SfMaps>
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tooltip.md:259]: Alignment is not proper.

File: MAUI/Maps/tooltip.md [hunk 11]
```diff
@@ -0,0 +1,468 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Maps tooltip appearance customization](images/tooltip/tooltip_textStyle.png)
+
+N>
+* The tooltip displays the binding context value for shapes and bubbles.
+* Marker tooltip displays a Latitude and Longitude value.
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tooltip.md:376]: The BindingContext of the Shape and Bubble will be the corresponding underline object.
The BindingContext of the Marker will be the corresponding MapMarker.

File: MAUI/Maps/tooltip.md [hunk 12]
```diff
@@ -0,0 +1,468 @@
...
+
+N>
+* Refer the [MapShapeLayer.ShapeTooltipTemplate](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeLayer/shapeTooltipBuilder.html), for customize tooltip for the shapes.
+* Refer the [MapShapeLayer.BubbleTooltipTemplate](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeLayer/bubbleTooltipBuilder.html), for customize tooltip for the bubbles.
+* Refer the [MapLayer.MarkerTooltipTemplate](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapLayer/markerTooltipBuilder.html), for customize tooltip for the markers.
+
+## Customizing Data Templates selector in ToolTip
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tooltip.md:383]: Use H3 heading.

Please refer to the below UG for descriptions
https://help.syncfusion.com/maui/listview/viewappearance

File: MAUI/Maps/tooltip.md [hunk 13]
```diff
@@ -0,0 +1,468 @@
...
+        {
+            layer.MarkerTooltipTemplate = Resources["LowPopulationTemplate"] as DataTemplate;
+        }
+    }    
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Maps tooltip appearance customization](images/tooltip/tooltip_selector.png)
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tooltip.md:463]: Image is not proper

File: MAUI/Maps/getting-started1.md [hunk 14]
```diff
@@ -0,0 +1,310 @@
...
+{% tabs %}
+
+{% highlight c# %}
+
+public MapSourcePage()
+{
+    InitializeComponent();
+    SfMaps map = new SfMaps();
+    MapShapeLayer layer = new MapShapeLayer();
+    layer.ShapesSource = MapSource.FromResource("australia.json");
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/getting-started1.md]: Noted was given sir

File: MAUI/Maps/getting-started1.md [hunk 15]
```diff
@@ -0,0 +1,310 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![maps basic view](images/getting-started/map_basic_view.png)
+
+## Mapping the data source for shape layer
+
+By default, the value specified for the [`ShapeDataField`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/shapeDataField.html) in the GeoJSON source will be used in the elements like data labels, tooltip, and legend for their respective shapes. However, it is possible to keep a data source and customize these elements based on the requirement. As mentioned above, [`ShapeDataField`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/shapeDataField.html) will be used to map with respective value returned in [`PrimaryValuePath`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/primaryValueMapper.html) from the data source.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/getting-started1.md]: Yes sir. Content changed

File: MAUI/Maps/getting-started1.md [hunk 16]
```diff
@@ -0,0 +1,310 @@
...
+* Refer the [`MapShapeLayer.BubbleSettings`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/bubbleSizeMapper.html), for customizing the bubble.
+* Refer the [`MapShapeLayer.DataLabelSettings`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/dataLabelMapper.html), for customizing the data label.
+* Refer the [`MapShapeLayer.ColorMappings`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/shapeColorValueMapper.html), for customizing the shape colors.
+
+## Add shape layer maps elements
+
+Add the basic maps elements such as data labels, legend, and tooltip as shown in the below code snippet.
+
+* **Data labels** - You can show data labels using the [`MapShapeLayer.ShowDataLabels`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeLayer/showDataLabels.html) property and also, it is possible to show data labels only for the particular shapes/or show custom text using the [`MapShapeLayer.DataLabelSettings`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/dataLabelMapper.html) property.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/getting-started1.md]: Modified sir

File: MAUI/Maps/getting-started1.md [hunk 17]
```diff
@@ -0,0 +1,310 @@
+---
+layout: post
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/getting-started1.md]: File name changed sir

File: MAUI/Maps/data-labels.md [hunk 18]
```diff
@@ -0,0 +1,294 @@
...
+
+You can show data labels on the map using the [`MapShapeLayer.ShowDataLabels`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeLayer/showDataLabels.html) property. By default, the data labels are rendered based on the value of [`ShapeDataField`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/shapeDataField.html) property. The default value of the [`ShowDataLabels`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeLayer/showDataLabels.html) property is `false`.
+
+{% tabs %}
+
+{% highlight xaml %}
+
+     <map:SfMaps>
+        <map:SfMaps.Layer>
+            <map:MapShapeLayer ShapesSource="https://cdn.syncfusion.com/maps/map-data/world-map.json"  DataSource="{Binding Data}" PrimaryValuePath="Continent" ShapeDataField="continent" ShowDataLabels = "True"  >
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/data-labels.md]: Modified sir

File: MAUI/Maps/data-labels.md (hunk 1 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/data-labels.md:91]: Changed sir

File: MAUI/Maps/shape.md [hunk 19]
```diff
@@ -0,0 +1,299 @@
...
+
+You can apply color to the shape based on whether the value returned from [`ColorMappings`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/shapeColorValueMapper.html) falls within the [`RangeColorMapping.From`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapColorMapper/from.html) and [`RangeColorMapping.To`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapColorMapper/to.html). Then, the [`RangeColorMapping.Color`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapColorMapper/color.html) will be applied to the respective shapes.
+
+{% tabs %}
+
+{% highlight xaml %}
+
+     <map:SfMaps>
+        <map:SfMaps.Layer>
+            <map:MapShapeLayer ShapesSource="https://cdn.syncfusion.com/maps/map-data/world-map.json" DataSource="{Binding Data}" PrimaryValuePath="Country" ShapeDataField="name" ShapeColorValuePath = "Storage" >
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/shape.md]: Changed sir

File: MAUI/Maps/legend.md [hunk 20]
```diff
@@ -0,0 +1,402 @@
...
+        }
+        public String Continent { get; set; }
+        public int Population { get; set; }
+    }
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Shape legend support](images/legend/default-legend.png)
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/legend.md:110]: Yes sir . Worked

File: MAUI/Maps/legend.md [hunk 21]
```diff
@@ -0,0 +1,402 @@
...
+documentation: ug
+---
+
+# Legend in MAUI Maps (SfMaps)
+
+You can provide clear information on the data plotted on the map using legend.
+
+## Shape legend
+
+You can show shape legend by setting the [`MapShapeLayer.Legend`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapLegend-class.html) property as `MapLegend(LegendSourceType.Shape)`. By default, the legend item's text is rendered based on the value of [`ShapeDataField`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapShapeSource/shapeDataField.html) property. The default value of the [`Legend`](https://pub.dev/documentation/syncfusion_flutter_maps/latest/maps/MapLegend-class.html) property is `null` and hence the legend will not be shown by default.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/legend.md]: Modified sir

File: MAUI/Maps/tooltip.md (hunk 9 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/tooltip.md:253]: Modified sir

File: MAUI/Maps/tooltip.md (hunk 10 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/tooltip.md:259]: Modified sir

File: MAUI/Maps/tooltip.md (hunk 12 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/tooltip.md:383]: Modified sir

File: MAUI/Maps/tooltip.md (hunk 13 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/tooltip.md:463]: Changed sir

**Pull request closed without merging**